packages/edit-navigation/src/components/inspector-additions/manage-locations.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useContext } from '@wordpress/element';
+import {
+	Spinner,
+	SelectControl,
+	CheckboxControl,
+	Button,
+	Modal,
+	PanelBody,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useMenuLocations, MenuIdContext } from '../../hooks';
+
+export default function ManageLocations( {
+	onSelectMenu,
+	isModalOpen,
+	openModal,
+	closeModal,
+} ) {
+	const menus = useSelect( ( select ) => select( 'core' ).getMenus(), [] );
+
+	const selectedMenuId = useContext( MenuIdContext );
+
+	const {
+		menuLocations,
+		assignMenuToLocation,
+		toggleMenuToLocation,
+	} = useMenuLocations();
+	const menusWithSelection = menuLocations.map( ( themeLocation ) => (
+		<CheckboxControl
+			key={ themeLocation.name }
+			checked={ themeLocation.menu === selectedMenuId }
+			onChange={ () =>
+				toggleMenuToLocation( themeLocation.name, selectedMenuId )
+			}
+			label={ themeLocation.name }
+		/>
+	) );
+	if ( ! menus || ! menuLocations ) {
+		return <Spinner />;
+	}
+
+	if ( ! menus.length ) {
+		return <p>{ __( 'There are no available menus.' ) }</p>;
+	}
```

Comment:
I believe this is now shown on a menu, so there has to be at least one menu for this component to render.